stl/inc/format

```diff
@@ -1827,9 +1827,10 @@ template <class _Handler>
 class _Specs_checker : public _Handler {
 private:
     _Basic_format_arg_type _Arg_type;
-    // we'll see this if we get a modifier that requires an integer presentation type
-    // for types that can have either integer or non-integer presentation types (charT or bool)
-    bool _Need_arithmetic_presentation_type = false;
+    // We'll see this if we get a modifier that requires an integer presentation type
+    // for types that can have either integer or non-integer presentation types (charT or bool).
+    bool _Need_arithmetic_presentation_type            = false;
+    bool _Need_arithmetic_or_pointer_presentation_type = false;

```

Comment:
@CaseyCarter asked during the weekly maintainer meeting whether adding this data member has ABI implications, but we concluded that it's safe because this internal type is used only to create local variables within functions, and is never directly exposed to users, nor does it create a situation where multiple functions need coordinated simultaneous changes.